Branch/BranchQRCode.m

```diff
@@ -0,0 +1,211 @@
+//
+//  BranchQRCode.m
+//  BranchMacOS
+//
+//  Created by Nipun Singh on 5/23/22.
+//  Copyright © 2022 Branch, Inc. All rights reserved.
+//
+
+#import <LinkPresentation/LPLinkMetadata.h>
+#import "BranchQRCode.h"
+#import "Branch.h"
+#import "BNCQRCodeCache.h"
+
+@implementation BranchQRCode
+
+NSString *buoTitle;
+CIImage *qrCodeImage;
+
+- (void) setMargin:(NSNumber *)margin {
+    if (margin.intValue > 20) {
+        margin = @(20);
+        BNCLogWarning(@"Margin was reduced to the maximum of 20.");
+    }
+    if (margin.intValue < 1) {
+        margin = @(0);
+        BNCLogWarning(@"Margin was increased to the minimum of 0.");
+    }
+    _margin = margin;
+}
+
+- (void) setWidth:(NSNumber *)width {
+    if (width.intValue > 2000) {
+        width = @(2000);
+        BNCLogWarning(@"Width was reduced to the maximum of 2000.");
+    }
+    if (width.intValue < 300) {
```

Comment:
Check should be for 500 instead of 300

Reply:
Updated the comment to reflect the minimum.